### PR TITLE
CCM-4956: Added supplierStatus information to GET message status endpoint documentation

### PR DESCRIPTION
**Pull request by DavidFueryBJSS**
## Summary
Added supplierStatus information to GET message status endpoint documentation

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Checklist
* [x] Brief description of work completed, and any technical decisions made as part of the PR
* [x] PR link added as a comment to the relevant JIRA ticket
* [x] PR link shared on Slack and/or Teams
* [ ] 2 reviews received
* [ ] Tester approval

### CODE REVIEW

File: specification/documentation/GetMessage.md [hunk 1]
```diff
@@ -34,6 +35,17 @@ Each channel can have one of the following statuses:
 * `delivered` - the channel has delivered the message
 * `failed` - the channel has failed to deliver the message
 
+Each channel can have one of the following supplier statuses:
```
Comment (aaron-downing) [inline specification/documentation/GetMessage.md:38]: We probably want to be explicit in saying these supplier statuses are specific to the NHSAPP channel

File: specification/documentation/GetMessage.md [hunk 2]
```diff
@@ -34,6 +35,17 @@ Each channel can have one of the following statuses:
...
+
+* `delivered` - the message has been successfully delivered to the user
+* `read` - a user has read the message
+* `notification_attempted` - a push notification is reported as having been sent to one or more device, but does not indicate whether the notifcation was received or displayed 
+* `unnotified` - it has been determined that a push notification has not been successfully relayed to any devices
+* `rejected` - the request to send the communication was rejected by the supplier
+* `notified` - a push notification is reported as having been successfully relayed to one or more devices
+* `received` - request has been received by the supplier and is queued to be processed
+* `unknown` - notify was unable to correctly determine the supplier status
```
Comment (aaron-downing) [inline specification/documentation/GetMessage.md:47]: `unknown - NHS notify was unable...`

File: specification/documentation/GetMessage.md (hunk 1 above)
Comment (aaron-downing) [reply specification/documentation/GetMessage.md:38]: Originally I think we said we'd have a section specifically for `Supplier Statuses` like we have for `Message Statuses` further down

File: specification/schemas/responses/MessageResponse.yaml [hunk 3]
```diff
@@ -56,6 +56,12 @@ properties:
                   type: string
                   description: If there is extra information associated with the status of this channel, it is provided here.
                   example: ""
+                supplierStatus:
+                  description: |-
+                    The current status of this message within the channel at the time this response was generated.
+      
+                    For more information please check our documentation on message & channel statuses above.
+                  $ref: ../enums/SupplierStatus.yaml
```
Comment (adadigital) [inline specification/schemas/responses/MessageResponse.yaml:64]: Will this always exist in each channel regardless of whether it's an NHSAPP channel?

File: specification/documentation/GetMessage.md [hunk 4]
```diff
@@ -34,6 +35,17 @@ Each channel can have one of the following statuses:
 * `delivered` - the channel has delivered the message
 * `failed` - the channel has failed to deliver the message
 
+Each channel can have one of the following supplier statuses:
+
+* `delivered` - the message has been successfully delivered to the user
+* `read` - a user has read the message
+* `notification_attempted` - a push notification is reported as having been sent to one or more device, but does not indicate whether the notifcation was received or displayed
```
Comment (tdroza-nhs) [inline specification/documentation/GetMessage.md:42]: "device" should be plural, and a typo on "notification"
```suggestion
* `notification_attempted` - a push notification is reported as having been sent to one or more devices, but does not indicate whether the notification was received or displayed 
```

File: specification/documentation/GetMessage.md [hunk 5]
```diff
@@ -34,6 +35,17 @@ Each channel can have one of the following statuses:
...
 
+Each channel can have one of the following supplier statuses:
+
+* `delivered` - the message has been successfully delivered to the user
+* `read` - a user has read the message
+* `notification_attempted` - a push notification is reported as having been sent to one or more device, but does not indicate whether the notifcation was received or displayed 
+* `unnotified` - it has been determined that a push notification has not been successfully relayed to any devices
+* `rejected` - the request to send the communication was rejected by the supplier
+* `notified` - a push notification is reported as having been successfully relayed to one or more devices
+* `received` - request has been received by the supplier and is queued to be processed
```
Comment (tdroza-nhs) [inline specification/documentation/GetMessage.md:46]:
```suggestion
* `received` - the request has been received by the supplier and is queued to be processed
```

File: specification/schemas/responses/MessageResponse.yaml (hunk 3 above)
Comment (DavidFueryBJSS) [reply specification/schemas/responses/MessageResponse.yaml:64]: After the next epic, yes ;)

**Review by adadigital — APPROVED** [review]

File: specification/documentation/GetMessage.md (hunk 2 above)
Comment (DavidFueryBJSS) [reply specification/documentation/GetMessage.md:47]: Done

File: specification/documentation/GetMessage.md [hunk 6]
```diff
@@ -22,6 +22,7 @@ Key values that are returned for each of these channels are:
 * `type` - the channel type
 * `channelStatus` - the status of that channel
 * `channelStatusDescription` - the channel status description
+* `supplierStatus` - the status of the message within that channel
```
[the remark this answers is not in the document]
Comment (DavidFueryBJSS) [reply specification/documentation/GetMessage.md:25]: Done

**Comment by DavidFueryBJSS** [comment]
image
Main change to structure in latest commit

**Review by aaron-downing — APPROVED** [review]

File: specification/schemas/enums/SupplierStatus.yaml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+title: Enum_SupplierStatus
+type: string
+enum:
+  - delivered
+  - read
+  - notification_attempted
+  - unnotified
+  - rejected
+  - notified
+  - received
```
Comment (aaron-downing) [inline specification/schemas/enums/SupplierStatus.yaml:10]: Missing `unknown`

Comment (DavidFueryBJSS) [reply specification/schemas/enums/SupplierStatus.yaml:10]: Added

**Pull request merged**